src/main/java/rx/exceptions/CompositeException.java

```diff
@@ -122,6 +122,7 @@ public synchronized Throwable getCause() {
                     // ignore
                     // the javadocs say that some Throwables (depending on how they're made) will never
                     // let me call initCause without blowing up even if it returns null
+                    chain = e;
                 }
                 chain = chain.getCause();
```

Comment:
I think this line should be moved into the `try` clause. Otherwise, `chain` will be assigned twice here.